redisinsight/ui/src/pages/home/components/DatabaseAlias/DatabaseAlias.tsx

```diff
@@ -77,76 +80,91 @@ const DatabaseAlias = (props: Props) => {
+                    className={styles.tooltipLogo}
+                    data-testid="tooltip-redis-stack-icon"
                   />
-                </InlineItemEditor>
-                <p className={styles.hiddenText}>{value}</p>
-              </>
+                )}
+                position="bottom"
+              >
+                <EuiIcon
+                  type={theme === Theme.Dark ? RediStackDarkMin : RediStackLightMin}
+                  className={styles.redistackIcon}
+                  data-testid="redis-stack-icon"
+                />
+              </EuiToolTip>
             </EuiFlexItem>
-          </EuiFlexGrid>
-        ) : (
-          <EuiText
-            className={styles.alias}
+          )}
+          <EuiFlexItem
+            onMouseEnter={onMouseEnterAlias}
+            onMouseLeave={onMouseLeaveAlias}
+            onClick={setEditMode}
+            grow
+            data-testid="edit-alias-btn"
+            style={{ overflow: isEditing || isHovering ? 'inherit' : 'hidden', maxWidth: '360px' }}
           >
-            <b className={styles.aliasText}>
-              <span>{alias}</span>
-            </b>
-            <b>
-              {database ? `[${database}]` : ''}
-            </b>
-          </EuiText>
-        )}
+            {isEditing || isHovering || isLoading ? (
+              <EuiFlexGrid
+                responsive
+                className="relative"
+                gutterSize="none"
+              >
+                <EuiFlexItem
+                  grow={1}
+                  component="span"
+                  className="fluid"
+                >
+                  <>
+                    <InlineItemEditor
+                      onApply={handleApplyChanges}
+                      onDecline={handleDeclineChanges}
+                      viewChildrenMode={!isEditing}
+                      isLoading={isLoading}
+                      isDisabled={!value}
+                      declineOnUnmount={false}
+                    >
+                      <EuiFieldText
+                        name="alias"
+                        id="alias"
+                        className={cx(styles.input)}
+                        placeholder="Enter Database Alias"
+                        value={value}
+                        fullWidth={false}
+                        maxLength={500}
+                        compressed
+                        isLoading={isLoading}
+                        onChange={onChange}
+                        append={!isEditing
+                          ? <EuiIcon type="pencil" color="subdued" /> : ''}
+                        autoComplete="off"
+                        data-testid="alias-input"
+                      />
+                    </InlineItemEditor>
+                    <p className={styles.hiddenText}>{value}</p>
+                  </>
+                </EuiFlexItem>
+              </EuiFlexGrid>
+            ) : (
+              <EuiText
```

Comment:
could you compress this component to 1 line?

Reply:
fixed